api/tests/unit/task_processor/test_unit_task_processor_processor.py

```diff
@@ -89,17 +89,28 @@ def _create_organisation(organisation_name):
 
     # When
     first_task_runs = run_recurring_tasks()
-    time.sleep(0.2)
 
-    second_task_runs = run_recurring_tasks()
+    # run the process again before the task is scheduled to run again to ensure
+    # that tasks are unlocked when they are picked up by the task processor but
+    # not executed.
+    no_task_runs = run_recurring_tasks()
 
-    task_runs = first_task_runs + second_task_runs
+    time.sleep(0.3)
```

Comment:
Is there a specific state we're waiting for here?

Reply:
Yeah, so the short answer is that we need to ensure that run_every (in this case 200ms) has passed.

The long answer is that this test should now ensure that the following happens:

first execution of run_recurring_tasks() - task has not been executed before so it should execute it
second execution of run_recurring_tasks() - task has been executed already and run_every (in this case 200ms) has not passed since the first run
third execution of run_recurring_tasks() - task has been executed already and run_every has passed since the first run.

Reply:
Both is good but just the one we have is pretty sufficient IMO.